client/src/js/services/ModalService.js

```diff
@@ -69,5 +74,80 @@ function ModalService(Modal) {
     });
   }
 
+  /**
+   * Select cashbox modal
+   */
+  function openSelectCashbox(request) {
+    var cashboxIsSet = request.cashbox && request.cashbox.id;
```

Comment:
This is quite confusing. What should `request` have in it?  Why would it have `request.cashboxId` and `request.cashbox.id`?